src/controllers/main-controller.ts

```diff
@@ -169,6 +170,68 @@ router.get(
   }),
 );
 
+router.get(
+  '/tickers',
+  asyncRoute(async (req: Request, res: Response) => {
+    const chainId = validateChainId(req, true);
+
+    const ammPools = await db
+      .select({
+        poolId: tAmmPools.pool,
+        baseTokenId: tAmmPools.token1Id,
+        quoteTokenId: tAmmPools.token2Id,
+        baseVolume: tAmmPools.token1Volume,
+        quoteVolume: tAmmPools.token2Volume,
+        liquidityInUsd: tAmmPools.token1Volume,
+      })
+      .from(tAmmPools)
+      .where(chainId ? eq(tAmmPools.chainId, chainId) : undefined)
+      .execute();
+
+    if (!ammPools.length) {
+      throw new BadRequestError('No pools found for the given chain.');
+    }
+
+    const tickers = await Promise.all(
+      ammPools.map(async (pool) => {
+        const baseToken = await db
+          .select({
+            symbol: tokens.symbol,
+            address: tokens.address,
+          })
+          .from(tokens)
+          .where(eq(tokens.id, pool.baseTokenId))
+          .limit(1)
+          .execute();
+
+        const quoteToken = await db
+          .select({
+            symbol: tokens.symbol,
+            address: tokens.address,
+          })
+          .from(tokens)
+          .where(eq(tokens.id, pool.quoteTokenId))
+          .limit(1)
+          .execute();
```

Comment:
this should be put to ammPools query as joins, it's now has 2 extra queries for each pool instead of having single one in total